src/kubeobjects/istio.go

```diff
@@ -0,0 +1,237 @@
+package kubeobjects
```

Comment:
not a fan of having all this stuff in `kubeobjects`, this package is for stuff that are "native" to kubernetes, which is not really true for istio 😄 

all of these could fit in the istio reconciler's package, because they are not used anywhere else (with a few exceptions, and even those don't need to be here)

Reply:
I will leave a bunch of comments where you could put these functions inside the `istio` package, you don't have to follow them, they will be just there to help you with naming and stuff 😉